coral-trino/src/main/java/com/linkedin/coral/trino/rel2trino/transformers/FromUtcTimestampOperatorTransformer.java

```diff
@@ -0,0 +1,133 @@
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.apache.calcite.rel.type.RelDataType;
+import org.apache.calcite.sql.SqlBasicTypeNameSpec;
+import org.apache.calcite.sql.SqlCall;
+import org.apache.calcite.sql.SqlDataTypeSpec;
+import org.apache.calcite.sql.SqlNode;
+import org.apache.calcite.sql.SqlOperator;
+import org.apache.calcite.sql.SqlTypeNameSpec;
+import org.apache.calcite.sql.fun.SqlStdOperatorTable;
+import org.apache.calcite.sql.type.BasicSqlType;
+import org.apache.calcite.sql.type.SqlTypeName;
+
+import com.linkedin.coral.common.HiveTypeSystem;
+import com.linkedin.coral.common.calcite.CalciteUtil;
+import com.linkedin.coral.common.transformers.SqlCallTransformer;
+import com.linkedin.coral.common.utils.TypeDerivationUtil;
+
+import static org.apache.calcite.sql.fun.SqlStdOperatorTable.*;
+import static org.apache.calcite.sql.parser.SqlParserPos.*;
+import static org.apache.calcite.sql.type.ReturnTypes.*;
+import static org.apache.calcite.sql.type.SqlTypeName.*;
+import static org.apache.calcite.sql.type.SqlTypeName.BIGINT;
+import static org.apache.calcite.sql.type.SqlTypeName.DATE;
+import static org.apache.calcite.sql.type.SqlTypeName.DOUBLE;
+import static org.apache.calcite.sql.type.SqlTypeName.INTEGER;
+
+
+/**
+ * This transformer operates on SqlCalls with 'FROM_UTC_TIMESTAMP(x, timezone)' Coral IR function
+ * and transforms it into different trino engine compatible functions based on the data type of 'x'.
+ * For Example:
+ * A SqlCall of the form: "FROM_UTC_TIMESTAMP(10000, 'America/Los_Angeles')" is transformed to
+ * "CAST(AT_TIMEZONE(FROM_UNIXTIME_NANOS(CAST(10000 AS BIGINT) * 1000000), $CANONICALIZE_HIVE_TIMEZONE_ID('America/Los_Angeles')) AS TIMESTAMP(3))"
+ */
+public class FromUtcTimestampOperatorTransformer extends SqlCallTransformer {
+
+  private static final Set<SqlTypeName> INT_SQL_TYPE_NAMES =
+      new HashSet<>(Arrays.asList(BIGINT, INTEGER, SMALLINT, TINYINT));
+  private static final Set<SqlTypeName> DECIMAL_SQL_TYPE_NAMES = new HashSet<>(Arrays.asList(DOUBLE, FLOAT, DECIMAL));
+  private static final Set<SqlTypeName> TIMESTAMP_SQL_TYPE_NAMES = new HashSet<>(Arrays.asList(TIMESTAMP, DATE));
+  private static final String FROM_UTC_TIMESTAMP = "from_utc_timestamp";
+  private static final String AT_TIMEZONE = "at_timezone";
+  private static final String WITH_TIMEZONE = "with_timezone";
+  private static final String TO_UNIXTIME = "to_unixtime";
+  private static final String FROM_UNIXTIME_NANOS = "from_unixtime_nanos";
+  private static final String FROM_UNIXTIME = "from_unixtime";
+  private static final String CANONICALIZE_HIVE_TIMEZONE_ID = "$canonicalize_hive_timezone_id";
+
+  public FromUtcTimestampOperatorTransformer(TypeDerivationUtil typeDerivationUtil) {
+    super(typeDerivationUtil);
+  }
+
+  @Override
+  protected boolean condition(SqlCall sqlCall) {
+    return sqlCall.getOperator().getName().equalsIgnoreCase(FROM_UTC_TIMESTAMP);
```

Comment:
Is it better to validate the number of the operands here, as in the function of `transform` directly access the two operands?

Reply:
operand count is also validated from the function signature defined in the [Registry](https://github.com/linkedin/coral/blob/6a5d8f229d809856efebe5ab93da972147732f7b/coral-hive/src/main/java/com/linkedin/coral/hive/hive2rel/functions/StaticHiveFunctionRegistry.java#L338). If the operand count is not 2, it should fail prior to this step.